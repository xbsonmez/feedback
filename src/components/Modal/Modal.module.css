.modal {
    position: absolute;
    top: 48%;
    left: 50%;
    background-color:transparent;
    transform: translate(-50%, -50%);
    min-width: 265px;
    width: calc(100% - 30px);
    max-width: 738px;
    max-height: calc(100% - 130px);
    border: 1px solid;
    z-index: 99;
    -webkit-overflow-scrolling: touch;
    min-height: 400px;
}

.modalError {
    position: absolute;
    top: 48%;
    left: 50%;
    background-color:transparent;
    transform: translate(-50%, -50%);
    min-width: 265px;
    width: calc(100% - 30px);
    max-width: 738px;
    max-height: calc(100% - 130px);
    border: 1px solid;
    z-index: 99;
    -webkit-overflow-scrolling: touch;
    min-height: 400px;
    color: var(--tyOrange);
}

.title {
    font-size: 25px;
    margin-top: 25px;
    margin-bottom: 25px;
    text-align: center;
}

.children {
    padding:20px;
    align-items: center;
    display: flex;
    flex-direction: column;
}

.closeBtn {
    position: absolute;
    width: 32px;
    height: 32px;
    right: 5px;
    cursor: pointer;
    background-color: transparent;
    border: none;
}