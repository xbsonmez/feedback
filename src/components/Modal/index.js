import React from 'react';
import ReactModal from 'react-modal';
import PropTypes from 'prop-types';
import {CloseOutlined } from '@ant-design/icons';
import s from './Modal.module.css';

if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#root');

const Modal = (props) => {
    const {children, isOpen, onClose, title, type} = props;

    return (
      <div data-testid="modal">
        <ReactModal
        isOpen={ isOpen }
        onRequestClose={ onClose }
        contentLabel="Modal"
        ariaHideApp={false}
        className={type ==='error' ? s.modalError : s.modal}
      >
       <button aria-label='close' className={s.closeBtn} onClick={ onClose } > <CloseOutlined /> </button>
        <div className={s.children}>
          <div data-testid="title" className={s.title}>
              {title}
          </div>
          { children }
        </div>
      </ReactModal>
      </div>
    );
  }
  
  Modal.propTypes = {
    isOpen: PropTypes.bool,
    onClose: PropTypes.func,
    children: PropTypes.node,
    title: PropTypes.string,
    type: PropTypes.string,
  };
  
  export default Modal;
  

