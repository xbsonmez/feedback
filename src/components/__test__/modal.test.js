import React from "react";
import {render, cleanup, screen, fireEvent} from '@testing-library/react';
import Modal from "../Modal";
import '@testing-library/jest-dom';

describe('Modal Component Test With React Testing Library', () => {

  afterEach(cleanup);

  const initialProps = {
    onClose: jest.fn(),
    children: "test child",
    title: "test title",
    isOpen:true
  };
  
    it('Should Take A Snapshot', () => {
      const { asFragment } = render(<Modal { ...initialProps }/>)
      expect(asFragment(<Modal { ...initialProps }/>)).toMatchSnapshot();
     });
     
     it('Click Close Event', () => {
      const handleClick = jest.fn();
      render(<Modal isOpen={true} onClose={handleClick}/>);
      const btn = screen.getByRole('button', {
        name: /close/i
      });
      fireEvent.click(btn);
      expect(handleClick).toHaveBeenCalledTimes(1);
     });

     it('Test Title', () => {
       const inputTitle = 'test';
      render(<Modal isOpen={true} title={inputTitle}/>);
      expect(screen.getByText(`test`)).toBeInTheDocument();
     });

     it('Test Childeren', () => {
      const inputChildren = 'test';
     render(<Modal isOpen={true} children={inputChildren}/>);
     expect(screen.getByText(`test`)).toBeInTheDocument();
    });

});