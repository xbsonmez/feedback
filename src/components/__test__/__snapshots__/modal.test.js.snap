// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Test With React Testing Library Should Take A Snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="modal"
  />
</DocumentFragment>
`;
